### DISCUSSION TITLE
"Show full" function in the Comparator

### DISCUSSION THREAD

**Discussion by sc818** [artisan-roaster-scope/artisan#788] [Ideas]
Hi! Marko and all the Artisan development team members:

I'd like to first say thank you for all the great works you've done. 
I'd like to know if it's possible to add a feature in the Comparator to "Show full"?
The reason is I track my BBP and have the program "Always ON".
So It would be really great if we can see the BBP data in the comparator.

Thank you very much!

**Comment by MAKOMO** [comment]
Excellent suggestion! I added a "BBP" flag to the comparator dialog. If ticked, the full recording before CHARGE is shown (not the part after DROP as this is not of much interest here and might just mess up the display). If you have the Time axis on AUTO in the Config >> Axis dialog, the timeaxis is automatically extended to the left such that all profile data is displayed, including those BBP parts.

Is this resolving the issue?

<img width="1183" alt="Screenshot 2022-01-28 at 17 05 55" src="https://user-images.githubusercontent.com/13909011/151581543-47d6a3f2-b89a-49b0-86b2-8de2139ad29b.png">

**Comment by sc818** [comment]
This is absolutely a dream come true!
Thank you for taking time for adding this great feature!
Much appreciated!

**Comment by sc818** [comment]
Dear Marko:
![011](https://user-images.githubusercontent.com/78765196/151690946-10f84364-3338-4f8d-adaf-8245750dadf1.png)

So I downloaded the beta continuous build 2.4.7 and try.
Now the comparator with the BBP view is excellent!
But in the time axis, if I set it the step to 1 minute, it will only show -2 minute minimum.
Would it be able to show in full time?

Thank you very much

**Reply by MAKOMO** [reply]
Could you please zip those two .alog files and attache them here on GitHub via drag-and-drop for me to investigate? Thanks!

**Reply by sc818** [reply]
Thank you Marko! I've uploaded the zip file.
[2002.1.31.zip](https://github.com/artisan-roaster-scope/artisan/files/7969051/2002.1.31.zip)

**Reply by MAKOMO** [reply]
Thanks for those files, I fixed this issues. New continuous builds with the fix are online.

**Reply by sc818** [reply]
Marko, I can’t thank you enough! Now it's just perfect!!!
